plugins/TagFix_Brand.py

```diff
@@ -0,0 +1,109 @@
+
+        self.errors[31301] = self.def_class(item=3130, level=2, tags=['brand', 'fix:chair'],
+                                            title=T_(
+                                                "This name is very common, is it a brand ?"),
+                                            resource='https://nsi.guide/', **doc)
+
+        self.country_code = self.father.config.options.get("country").split("-")[0]
+        self.frequent_names_from_nsi = self._get_frequent_names()
+        self.brands_from_nsi = self._get_brands()
+
+    def _get_frequent_names(self):
+        nsi_url_for_names = "https://raw.githubusercontent.com/osmlab/name-suggestion-index/master/dist/names_keep.json"
+        json_str = urlread(nsi_url_for_names, 30)
+        results = json.loads(json_str)
+        return [elem.lower() for elem in results.keys()]
+
+    def _get_brands(self):
+        nsi_url_for_brands = "https://raw.githubusercontent.com/osmlab/name-suggestion-index/master/dist/brands.json"
+        json_str = urlread(nsi_url_for_brands, 30)
+        results = json.loads(json_str)
+        additional_brands = {}
+        for brand_nsi_name, brand in results["brands"].items():
+            if "countryCodes" in brand and self.country_code.lower() not in brand["countryCodes"]:
+                #print("ignoring brand " + brand_nsi_name)
+                continue
+            brand_nsi_name = brand_nsi_name.split("~")[0]
+            if "matchTags" in brand:
+                for additional_tag in brand["matchTags"]:
+                    nsi_key = "{}|{}".format(additional_tag, brand_nsi_name.split("|")[1])
+                    additional_brands[nsi_key.lower()] = brand
+            if "matchNames" in brand:
+                for additional_name in brand["matchNames"]:
+                    nsi_key = "{}|{}".format(brand_nsi_name.split("|")[0], additional_name)
+                    additional_brands[nsi_key.lower()] = brand
+            additional_brands[brand_nsi_name.lower()] = brand
+        return additional_brands
+
+    def node(self, data, tags):
+        if "name" in tags and not "brand" in tags:
+            for main_key in ["shop", "amenity"]:
+                if main_key in tags:
+                    nsi_key = "{}/{}|{}".format(main_key, tags[main_key], tags["name"]).lower()
+                    if nsi_key in self.frequent_names_from_nsi:
+                        if nsi_key in self.brands_from_nsi.keys():
+                            brands_tags = self.brands_from_nsi[nsi_key]["tags"]
+                            tags_to_add = {}
+                            for tag in brands_tags:
+                                if not tags.get(tag):
+                                    tags_to_add[tag] = brands_tags[tag]
+                            return {"class": 31301, "subclass": 0,
+                                    "fix": {"+": tags_to_add}}
```

Comment:
No sure here. If the tag already exist you need to use '~' in place of '+'.

Reply:
It does not already exist (`if not tags.get(tag)`) : it only add tags that are not present yet.

Do you want to change that and replace all existing keys and values by the ones in the nsi ?  It seems a bit excessive to me but this is arguable.

Reply:
Yes no replace. Ok.